### PR TITLE
Travis

### PR DESCRIPTION
**Pull request by jlconlin**

### CODE REVIEW

File: implementation/Table.hpp.in [hunk 1]
```diff
@@ -6,6 +6,19 @@
 #include <type_traits>
 #include <vector>
 
+namespace ACEtk {
+namespace implementation {
+
+template< typename T >  constexpr const char* name = "";
```
Comment (jlconlin) [inline implementation/Table.hpp.in]: I like this a lot better than the macro below. It's easier to read.

Comment (apmccartney) [inline implementation/Table.hpp.in]: C++14 variable templates are pretty useful for generic programming shenanigans and one step lower than macros on the heirarchy of evil.

Comment (jlconlin) [inline implementation/Table.hpp.in]: I'm still not sure why we need the variable templates instead of declaring the various arrays explicitly.

Comment (apmccartney) [inline implementation/Table.hpp.in]: Did I ever claimed it was necessary? There are lots of ways to skin a cat.

What advantages are there to declaring them explicitly?

Comment (jlconlin) [inline implementation/Table.hpp.in]: Even easier to read.

Comment (apmccartney) [inline implementation/Table.hpp.in]: Really? Consider the context that the variable is actually [used.](https://github.com/njoy/ACEtk/blob/6c43af7ccc656c34ae8b6646036ca1262eb30607/implementation/Table/src/verifyIndex.hpp.in#L26)

Let's consider two alternatives: the existing implementation and something along the lines of your suggestion.
```cpp

template< typename Sequence  >
static void
verifyIndex( const Sequence& sequence, const std::size_t index ) {
  
  if ( ( index < 1 ) || ( index > sequence.size() ) ){
    LOG(ERROR) << "Illegal index argument";
    LOG(INFO) << "Index value: " << index;
    LOG(INFO) << name< Sequence >
              << " may be indexed with values between 1 and "
              << sequence.size() << " inclusively";
    throw std::out_of_range( "" );
  }
}
```

```cpp

template< typename Sequence  >
static void
verifyIndex( const Sequence& sequence, const std::size_t index, const std::string& name ) {
  
  if ( ( index < 1 ) || ( index > sequence.size() ) ){
    LOG(ERROR) << "Illegal index argument";
    LOG(INFO) << "Index value: " << index;
    LOG(INFO) << name << " may be indexed with values between 1 and "
              << sequence.size() << " inclusively";
    throw std::out_of_range( "" );
  }
}
```
The intent isn't any more or less clear in either implementation in my opinion.

**Pull request merged**